src/main/java/hudson/remoting/Channel.java

```diff
@@ -832,9 +834,15 @@ public void setJarCache(@Nonnull JarCache jarCache) {
      */
     public <V,T extends Throwable>
     V call(Callable<V,T> callable) throws IOException, T, InterruptedException {
+        return call(callable, UserRequest.DEFAULT_PERFORM_TIMEOUT, UserRequest.DEFAULT_EXECUTION_TIMEOUT);
+    }
+    
+    public <V,T extends Throwable>
+    V call(@Nonnull Callable<V,T> callable, @CheckForNull Duration performTimeout, @CheckForNull Duration executionTimeout) 
```

Comment:
I do not think `executionTimeout` is necessary. Just keep it to `performTimeout`. No one cares about the difference. A caller simply wants to ensure that a network call does not block forever.